internal/broker/broker.go

```diff
@@ -433,7 +432,12 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
 			return AuthDenied, errorMessage{Message: "could not get required response"}
 		}
 
-		t, err := session.authCfg.oauth.DeviceAccessToken(ctx, response, b.providerInfo.AuthOptions()...)
+		if response.Expiry.IsZero() {
+			response.Expiry = time.Now().Add(24 * time.Hour)
```

Comment:
You are being very generous :) I think less is acceptable to have hanging connection. Probably something like an hour is more than enough and will free up resources if this hangs :p

Why this is not `maxRequestTime` btw?

Reply:
This takes a while since we generate the code when the user selects the device authentication mode. Considering the time he would take to open the browser, type everything, and so on... It can't be a short time. Also, when it's zero it technically does not expire, that's why I went for a day (I could replace the context instead, but I think this is simpler).

Reply:
Right, but I’m just thinking about requests piling up. And it won't take a full day to open a browser, let’s make in an hour and not more, if you don’t mind?

Reply:
Yeah, sure... It doesn't matter, tbh, but I will change it to an hour instead.